langlib/langlib-test/src/test/resources/test-src/functionlib_test.bal

```diff
@@ -0,0 +1,254 @@
+    return person.firstName + middleName + person.secondName;
+}
+
+function test11(*NewStudent person) returns string {
+    return person.firstName + " " + person.secondName;
+}
+
+function test12(*Foo foo, *Bar bar) returns anydata {
+    return foo["value"];
+}
+
+function test13(int a, int b = 5, *Foo foo, int... c) returns int {
+    return a + b;
+}
+
+function test14(int a, int b, *Foo foo) returns int {
+    return a + b + <int> foo["value"];
+}
+
+function test15(int a = 10, int b = 15, *Foo foo) returns int {
+    return a + b + <int> foo["value"];
+}
+
+function testCallFunctionWithIncludedRecordParameters() {
+    NewPerson person = {firstName: "chiran", secondName: "sachintha"};
+    NewStudent student = {};
+    Foo f = {"value": 10};
+    Bar b = {a:10, b: 20};
+
+    assertEquality(function:call(test9, person), "chiran sachintha");
+    assertEquality(function:call(test10, " ", person), "chiran sachintha");
+    assertEquality(function:call(test11, student), "chiran sachintha");
+    assertEquality(function:call(test12, f, b), 10);
+    assertEquality(function:call(test13, 10, 10, f), 20);
+    assertEquality(function:call(test13, 10, 10, f, 10, 10), 20);
+    assertEquality(function:call(test14, 10, 10, f), 30);
+    assertEquality(function:call(test15, 15, 15, f), 40);
+}
+
+function test16() returns any|error {
+    function(int, int) returns int addFunction = func1;
+    return function:call(addFunction, 1, 2);
+}
+
+function func1(int a, int b) returns int {
+    return a + b;
+}
+
+function test17() returns any|error {
+    function(int, int) returns string sumFunction = function(int a, int b) returns string {
+                                                        int value =  a + b;
+                                                        return "sum is " + value.toString();
+                                                    };
+    return function:call(sumFunction, 1, 2);
+}
+
+function test18() returns any|error {
+    return function:call(func2, 1, func1);
+}
+
+function func2(int a, function(int x, int y) returns int func) returns int {
+    return  a + func(1, 2);
+}
+
+function test19() returns any|error {
+    function(string a, string b) returns string foo = func3();
+    return function:call(foo, "hello ", "world.");
+}
+
+function func3() returns function(string a, string b) returns string {
+                            return function(string x, string y) returns string {
+                                        return x + y;
+                                    };
+}
+
+function testCallFunctionWithFunctionPointers() {
+    assertEquality(test16(), 3);
+    assertEquality(test17(), "sum is 3");
+    assertEquality(test18(), 4);
+    assertEquality(test19(), "hello world.");
+}
+
+function testCallFunctionWithInvalidOutput() {
+    any|error a1 = trap function:call(value1, 10);
+    assertEquality(true, a1 is error);
+    assertEquality("{ballerina/lang.function}IncompatibleTypes", (<error> a1).message());
```

Comment:
Why aren't we asserting the detail message? How are we testing the error message without those assertions?

Reply:
With https://github.com/ballerina-platform/ballerina-lang/pull/36555/commits/75701f879491e2e3ea9ad5ca0e6acc791cbb3ef6 assert the detail message also